application/common/models/Invite.php

```diff
@@ -0,0 +1,103 @@
+<?php
+
+namespace common\models;
+
+use Closure;
+use common\helpers\MySqlDateTime;
+use Ramsey\Uuid\Uuid;
+use Yii;
+use yii\helpers\ArrayHelper;
+
+class Invite extends InviteBase
+{
+    public function rules(): array
+    {
+        return ArrayHelper::merge([
+            [
+                'uuid', 'default', 'value' => Uuid::uuid4()->toString()
+            ],
+            [
+                'created_utc', 'default', 'value' => MySqlDateTime::now(),
+            ],
+            [
+                'expires_on', 'default', 'value' => $this->expires(),
+            ],
+        ], parent::rules());
+    }
+
+    private function expires(): Closure
+    {
+        return function() {
+            $lifespan = Yii::$app->params['inviteLifespan'];
+
+            return MySqlDateTime::formatDate(strtotime($lifespan, strtotime($this->created_utc)));
+        };
+    }
+
+    public function attributeLabels(): array
+    {
+        $labels = parent::attributeLabels();
+
+        $labels['uuid'] = Yii::t('app', 'UUID');
+        $labels['created_utc'] = Yii::t('app', 'Created (UTC)');
+
+        return $labels;
+    }
+
+    /**
+     * Checks expiration date. If `expires_on` is today or before, `isValidCode`
+     * returns `false` and adds an error message.
+     *
+     * @return bool
+     * @throws \Exception
+     */
+    public function isValidCode()
+    {
+        $expiration = strtotime($this->expires_on);
+        if ($expiration === false) {
+            throw new \Exception('Unable to parse expires_on');
+        }
+
+        $now = time();
+        if ($now > $expiration) {
+            $this->addError('expires_on', 'Expired code.');
+            return false;
+        }
+        return true;
+    }
+
+    /**
+     * Return an existing, non-expired invite code, or create a new object and
+     * return the new code.
+     *
+     * @param int $userId
+     * @return Invite
+     * @throws \Exception
+     */
+    public static function getInviteCode(int $userId): Invite
```

Comment:
This method's name doesn't quite seem to line up with what it actually does: it says it will get an invite code, but it actually returns an Invite instance. I'm not sure which of those you'll end up wanting a function for (possibly both), and that will probably guide which thing you change here.

Reply:
Done.